packages/shared/src/hooks/utils/useEasterEggTheme.ts

```diff
@@ -0,0 +1,67 @@
+import { useLayoutEffect, useMemo } from 'react';
+import { useRouter } from 'next/router';
+import { ThemeMode, useSettingsContext } from '../../contexts/SettingsContext';
+import { useFeature } from '../../components/GrowthBookProvider';
+import { feature } from '../../lib/featureManagement';
+
+interface UseEasterEggThemeResult {
+  version: number;
+  logo?: string;
+  logoText?: string;
+  body?: Record<string, string>;
+  navbar?: string;
+  cursor?: string;
+}
+
+export const useEasterEggTheme = (): UseEasterEggThemeResult | undefined => {
+  const { themeMode } = useSettingsContext();
+  const easterEggTheme = useFeature(feature.easterEggTheme);
+  const router = useRouter();
+  const isOnboarding = router?.pathname?.startsWith('/onboarding') ?? false;
+  const useTheme = !isOnboarding && !!easterEggTheme;
+
+  const isLight = useMemo(() => {
+    if (themeMode === ThemeMode.Auto) {
+      return globalThis?.window?.matchMedia?.('(prefers-color-scheme:light)')
+        .matches;
+    }
+
+    return themeMode === ThemeMode.Light;
+  }, [themeMode]);
+  const theme = isLight ? ThemeMode.Light : ThemeMode.Dark;
+
+  useLayoutEffect(() => {
+    const elem = globalThis?.document?.body;
+
+    // reset body styles
+    if (elem?.style?.cssText) {
+      elem.style.cssText = '';
+    }
+
+    if (!useTheme) {
+      return;
+    }
+
+    if (easterEggTheme?.[theme]?.body) {
+      Object.entries(easterEggTheme[theme]?.body).forEach(([key, value]) => {
+        elem?.style?.setProperty(key, value);
+      });
+    }
```

Comment:
Maybe more efficient solution would be to compute style tag and then set it to `elem.style` in one go. Like this you get a global repaint on each key iteration which can be quite taxing.

Reply:
Good idea, let me give it a try.

Reply:
Switched to adding a stylesheet element to the page head instead. Also, the content of the stylesheet is compared to the existing stylesheet added previously, so we do not replace the existing stylesheet with the same one, avoiding repaint of the same styles.